src/Grid/grid.jl

```diff
@@ -84,6 +205,8 @@ mutable struct Grid{dim,C<:AbstractCell,T<:Real} <: AbstractGrid{dim}
     vertexsets::Dict{String,Set{VertexIndex}} 
     # Boundary matrix (faces per cell × cell)
     boundary_matrix::SparseMatrixCSC{Bool,Int}
+    #topology
+    topology::topologytype
```

Comment:
I think adding some abstraction to access the topology information could be helpful to not leak implementation details, i.e. something like

Get everything connected to an entity
* `full_neighborhood(::Grid, ::CellIndex, include_self=false)`
* `full_neighborhood(::Grid, ::FaceIndex, include_self=false)`
* `full_neighborhood(::Grid, ::EdgeIndex, include_self=false)`
* `full_neighborhood(::Grid, ::VertexIndex, include_self=false)`

and maybe some specialized methods for relations which can not be directly derived, e.g.

* `face_neighborhood(::Grid, ::CellIndex, include_self=false)`

I think this should be sufficient to quickly get all topological connections, but I am not sure. I think checking this is out of the scope of this PR and more future work.